contracts/PolymorphWithGeneChanger.sol

```diff
@@ -62,13 +76,15 @@ contract PolymorphWithGeneChanger is IPolymorphWithGeneChanger, Polymorph {
 
     function priceForGenomeChange(uint256 tokenId) public override virtual view returns(uint256 price) {
         uint256 pastChanges = _genomeChanges[tokenId];
+        require(pastChanges < 255, 'pastChanges is too big and will cause overflow.');
+
         price = baseGenomeChangePrice;
         
         for(uint256 i = 0; i < pastChanges; i++) {
```

Comment:
I think here the proposed implementation is without the for loop. Like this:
    
    function priceForGenomeChange(uint256 tokenId) public override virtual view returns(uint256 price) {
        uint256 pastChanges = _genomeChanges[tokenId];
        return baseGenomeChangePrice.mul(1 << pastChanges);
    }
    Can you test it and @Perseverance can confirm as well :)